### PR TITLE
Add mark and unmark functionality.

### PR DESCRIPTION
**Pull request by doimoiboi**
#40

### CODE REVIEW

**Review by yiyefyy — APPROVED** [review]
Amazinggg

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -105,12 +105,22 @@ public void addPerson(Task task) {
...
+    @Override
+    public void markTask(Task target) {
+        addressBook.markTask(target);
+    }
+
+    @Override
+    public void unmarkTask(Task target) {
+        addressBook.unmarkTask(target);
+    }
+
```
Comment (XenonShawn) [inline src/main/java/seedu/address/model/ModelManager.java]: Should add in the `requireNonNull` check

File: src/main/java/seedu/address/model/task/Task.java [hunk 2]
```diff
@@ -19,7 +19,7 @@ public class Task {
     private final Module module;
     private final Deadline deadline;
     private final Set<Tag> tags = new HashSet<>();
-    private final boolean isDone;
+    private boolean isDone;
```
Comment (XenonShawn) [inline src/main/java/seedu/address/model/task/Task.java]: `Task` is supposed to be immutable.

**Review by XenonShawn — CHANGES_REQUESTED** [review]
`Task` must be immutable - a new Task object should be created every time it's changed.

**Review by XenonShawn — APPROVED** [review]
Works, and LGTM!

File: src/main/java/seedu/address/model/task/Task.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ public Set<Tag> getTags() {
         return Collections.unmodifiableSet(tags);
     }
 
-    public boolean isDone() {
+    public Boolean isDone() {
```
Comment (XenonShawn) [inline src/main/java/seedu/address/model/task/Task.java:65]: Should be a normal `boolean`

**Pull request merged**